### ISSUE TITLE
regression: invalid doc attribute

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#123293]
```

[INFO] [stdout] ---- src/lib.rs - (line 48) stdout ----
[INFO] [stdout] error: invalid `doc` attribute
[INFO] [stdout]   --> src/lib.rs:56:11
[INFO] [stdout]    |
[INFO] [stdout] 11 |     #[doc("This attribute will be applied on the `impl` block")] Left, Right:
[INFO] [stdout]    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[INFO] [stdout]    |
[INFO] [stdout]    = note: `#[deny(invalid_doc_attributes)]` on by default
[INFO] [stdout] 
[INFO] [stdout] error: invalid `doc` attribute
[INFO] [stdout]   --> src/lib.rs:57:11
[INFO] [stdout]    |
[INFO] [stdout] 12 |     #[doc("This attribute will be applied on the `fn` block")] a + b -> Answer {
[INFO] [stdout]    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[INFO] [stdout] 
[INFO] [stdout] error: aborting due to 2 previous errors
```

https://crater-reports.s3.amazonaws.com/beta-1.78-3/beta-2024-03-30/reg/operator-sugar-0.1.2/log.txt

**Comment by fmease** [comment]
I, as the reviewer, did a crater run for this in the regression PR #121545, found a single regression which was fixed by a contributor, and I deemed it okay to merge it as it's a correctness fix. I was fully aware that this is was a breaking change.

Since an incredibly similar PR authored by myself, namely #104148, was merged without a (lang) FCP and only a crater run more than a year ago, I followed suit.
